mods/gen1/moves.js

```diff
@@ -426,11 +425,9 @@ exports.BattleMovedex = {
 					if (pokemon.status === 'tox') {
 						pokemon.setStatus('psn');
 					}
-					let volatiles = Object.keys(pokemon.volatiles);
-					for (let n = 0; n < volatiles.length; n++) {
-						let id = volatiles[n];
+					for (const id of Object.keys(pokemon.volatiles)) {
 						if (id === 'residualdmg') {
-							pokemon.volatiles[id].counter = 0;
+							id.counter = 0;
```

Comment:
...it doesn't work this way

Reply:
shhhhhh

Reply:
I'm actually surprised eslint does- oh I keep forgetting eslint doesn't know what types are.

Reply:
If you TypeScripted this file, though, TypeScript would be yelling at you here.

Reply:
Today is a good day to learn about what happens with primitive promotion in JavaScript! In JavaScript, there are a bunch of primitive types, but the main relevant ones here are `boolean`, `string`, and `number`. `id` is a `string`, specifically the string `"residualdmg"`.

So here's the thing: Primitives don't actually have attributes. Only objects have attributes.

When you access attributes on a primitive, JavaScript will temporarily promote the primitive to an object. So when you write something like:

```js
id.counter = 0;
```

What's actually going on behind the scenes is:

```js
{
  let promotedId = new Object(id);
  promotedId.counter = 0;
} // promotedId is immediately destroyed; your code had no effect
```